classes/module/Module.php

```diff
@@ -953,32 +874,7 @@ public function registerHook($hook_name, $shop_list = null)
      */
     public function unregisterHook($hook_id, $shop_list = null)
     {
-        // Get hook id if a name is given as argument
-        if (!is_numeric($hook_id)) {
-            $hook_name = (string)$hook_id;
-            // Retrocompatibility
-            $hook_id = Hook::getIdByName($hook_name);
-            if (!$hook_id) {
-                return false;
-            }
-        } else {
-            $hook_name = Hook::getNameById((int)$hook_id);
-        }
-
-        Hook::exec('actionModuleUnRegisterHookBefore', array('object' => $this, 'hook_name' => $hook_name));
-
-        // Unregister module on hook by id
-        $sql = 'DELETE FROM `'._DB_PREFIX_.'hook_module`
-            WHERE `id_module` = '.(int)$this->id.' AND `id_hook` = '.(int)$hook_id
-            .(($shop_list) ? ' AND `id_shop` IN('.implode(', ', array_map('intval', $shop_list)).')' : '');
-        $result = Db::getInstance()->execute($sql);
-
-        // Clean modules position
-        $this->cleanPositions($hook_id, $shop_list);
-
-        Hook::exec('actionModuleUnRegisterHookAfter', array('object' => $this, 'hook_name' => $hook_name));
-
-        return $result;
+        return Hook::unregisterHook($this, $hook_id, $shop_list);
```

Comment:
I don't think we should deprecate it, every single module use it, it will be a loooot of notice. :smile: 
Plus it seems fine to ask the module to register itself.